app/controllers/api/v1/rubygems_controller.rb

```diff
@@ -1,13 +1,14 @@
 class Api::V1::RubygemsController < Api::BaseController
-  before_action :authenticate_with_api_key, only: %i[index create]
-  before_action :find_rubygem,              only: %i[show reverse_dependencies]
-
+  before_action :find_api_key, only: %i[index create]
```

Comment:
As above, prefer secure default of `except` and method name that reveals this performs authentication.